tests/unit/externalRequestsTest.js

```diff
@@ -81,13 +294,172 @@ tap.test('External Requests Middleware', {autoend: true}, t => {
+          t.type(err, Error)
+          t.equal(err.message, 'Rejected Promise: Fail')
         })
-      }
-    )
+      })
+
+      t.test(
+        'should add response data to the ctx when all promises resolve',
+        t => {
+          t.plan(1)
 
+          const performRequestsStub = externalRequests.__set__(
+            'performRequests',
+            () => [
+              Promise.resolve({test1: 'testA'}),
+              Promise.resolve({test2: 'testB'})
+            ]
+          )
+
+          const ctx = {
+            state: {
+              uuid: 'randomUidForRequest',
+              metaData: {
+                name: 'Testing endpoint',
+                requests: {
+                  lookup: [
+                    {
+                      // first lookup request config - responds with success
+                    },
+                    {
+                      // second lookup request config - responds with success
+                    }
+                  ]
+                }
+              }
+            }
+          }
+
+          const prepareLookupRequests = externalRequests.__get__(
+            'prepareLookupRequests'
+          )
+
+          prepareLookupRequests(ctx).then(() => {
+            t.same({test1: 'testA', test2: 'testB'}, ctx.lookupRequests)
+            performRequestsStub()
+          })
+        }
+      )
+      t.test(
+        "should send requests and set the response body's statusText to Completed",
+        async t => {
+          const url = 'http://localhost:8000/'
+          const method = 'PUT'
+          const id = 'Patient'
+
+          const ctx = {
+            status: 200,
+            state: {
+              metaData: {
+                requests: {
+                  response: [
+                    {
+                      config: {
+                        url: `${url}patient?name=raze`,
+                        method: method
+                      },
+                      id: id,
+                      primary: true
+                    }
+                  ]
+                }
+              }
+            },
+            request: {
+              header: {
+                'x-openhim-transactionid': '1232244'
+              }
+            },
+            response: {
+              headers: {}
+            },
+            body: {
+              surname: 'breez'
+            },
+            set: (key, value) => {
+              ctx.response.headers[key] = value
+            }
+          }
+
+          const response = {name: 'raze', surname: 'breez'}
+
+          nock(url)
+            .put('/patient?name=raze')
+            .reply(400, response)
+
+          await orchestrateMappingResult(ctx)
+
+          t.equals(ctx.orchestrations.length, 1)
+          t.match(ctx.body, /Completed/)
+          t.end()
+        }
+      )
+
+      t.test(
+        "should set the response body's statusText to 'Completed with error(s)' (error on non primary request",
```

Comment:
The description seems a bit incorrect with a missing closing bracket

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/39/commits/f6e854686999d065f57f20f9f1af3f9e1318f21e